osu.Game.Tests/Visual/Online/TestNowPlayingCommand.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps;
+using osu.Game.Online.Chat;
+using osu.Game.Rulesets;
+using osu.Game.Users;
+
+namespace osu.Game.Tests.Visual.Online
+{
+    [HeadlessTest]
+    public class TestNowPlayingCommand : OsuTestScene
+    {
+        [Cached(typeof(IChannelPostTarget))]
```

Comment:
unsure why this is required even though it's specified on the interface (not required for the game usage, just for nunit testing).

Reply:
I have debugged a bit through this and it seems that this is an edge case where having a property / field that is of a type that has `Cached` attribute will never be cached..., will look into fixing this framework-side if you haven't gone already.

Reply:
what you are saying makes no sense

Reply:
So the actual usage in the game works fine because the commands are children to the `ChannelManager` class with `IChannelPostTarget` interface assigned, while here in this test, the post target here is a property instead with type `PostTarget` that has `IChannelPostTarget` interface assigned to it.

There could be two solutions this:
 1. To solve this here by making the `PostTarget` class a `CompositeDrawable` and adding commands inside it.
 2. To solve this framework-side by making `CachedAttribute.CreateActivator(...)` aware of properties and fields that has a type with `CachedAttribute` assigned to it or the type's interfaces.

I'm a bit hesitant about the second solution as it may introduce in bugs with currently existing usages.

Reply:
the problem is the cached is not working. stating arbitrary workarounds is unnecessary.

Reply:
I'd say this is intended, fwiw. Caching on class/interface definitions respects the top-down model of the hierarchy, where the cached value is available for all its children.

Having it work as expected here means there would be a lateral component to it, where a child now caches itself in its parent for all other children of the parent to access. Which sounds wrong.

The usage here is correct in my books.

Reply:
I see. Might be worthwhile adding a mention of this to `[Cached]` (if it tripped me up, it's likely going to trip others up).